src/test/java/com/networknt/schema/regex/Issue814Test.java

```diff
@@ -8,40 +8,50 @@ class Issue814Test {
 
     @Test
     void jdkTypePattern() {
-        JDKRegularExpression ex = new JDKRegularExpression("^list|date|time|string|enum|int|double|long|boolean|number$");
+        JDKRegularExpression ex = new JDKRegularExpression("^(list|date|time|string|enum|int|double|long|boolean|number)$");
         assertTrue(ex.matches("list"));
         assertTrue(ex.matches("string"));
         assertTrue(ex.matches("boolean"));
         assertTrue(ex.matches("number"));
         assertTrue(ex.matches("enum"));
+        assertFalse(ex.matches("listZ"));
+        assertFalse(ex.matches("AenumZ"));
+        assertFalse(ex.matches("Anumber"));
     }
 
     @Test
     void jdkOptionsPattern() {
-        JDKRegularExpression ex = new JDKRegularExpression("^\\d*|[a-zA-Z_]+$");
+        JDKRegularExpression ex = new JDKRegularExpression("^\\d+|[a-zA-Z_]+$");
```

Comment:
Note that I changed `*` to `+`: because `^\d*` is vacuously true: it matches the empty string at the start of any input. (Without this the `internal9` test case below would also yield `true`.)